### ISSUE TITLE
Perform manual testing for Shuffle integration

### ISSUE THREAD

**Issue by fdalmaup** [wazuh/wazuh-qa#3398]
| Target version | Related issue | Related PR |
|--------------------|--------------------|-----------------|
|          4.4.0              |    [ #15034   ](https://github.com/wazuh/wazuh/issues/15034)                |  https://github.com/wazuh/wazuh/pull/15054                  |

## Description

In order to validate the recently added Shuffle integration, we need to perform the tests manually.

## Proposed checks

- [x] Verify the correct configuration of the integration
- [x] Verify the proper sending of alerts to the external platform using the webhook

## Expected results

Logs indicating each step of the integration process in `/var/ossec/logs/integrations.log`

## Configuration and considerations

It is needed to add the following configuration to `ossec.conf`
```
<integration>
     <name>shuffle</name>
     <hook_url>http://<IP>:3001/api/v1/hooks/<HOOK_ID></hook_url>
     <level>3</level>
     <alert_format>json</alert_format>
 </integration>
```
And restart Wazuh manager afterward.

**Comment by Rebits** [comment]
Tester review

| Tester | PR commit               | 
|--------|-------------------------|
| @Rebits  |  [848710b](https://github.com/wazuh/wazuh/pull/15054/commits/848710b0e130ba2d6ec961d514947d0d7795586b)

### Testing environment

| OS                  | OS version | Deployment | Image/AMI               |
| ------------------- | ---------- | ---------- | ----------------------- |
| Ubuntu              | 22         | EC2        | `ami-003530de8839921c4` |


### Tested packages

| OS     | Package                                                                             |
| ------ | ----------------------------------------------------------------------------------- |
| Ubuntu | [Manager](https://packages-dev.wazuh.com/warehouse/pullrequests/4.4/deb/var/wazuh-manager_4.4.0-0.commit848710b_amd64.deb)|


### Status

- [ ] In progress
- [ ] Pending Review
- [ ] QA team leader approved
- [ ] QA manager approved @jmv74211 
- [ ] Development team leader approved @72nomada

**Comment by jmv74211** [comment]
## QA review

- **Type**: Manual testing.
- **Status**: Requested improvements 🟡 
- **Comments**: We have found reasons for which we cannot approve the PR. These are as follows:
    - ***(1)* There is not enough logging to inform the user about possible failure cases or warnings that the integration is not working properly, even if it is a configuration or suffle part. (cc @wazuh/framework)**

        For example, if an alert does not reach shuffle because the response is other than 200, we believe it would be convenient to indicate it in the log beyond the DEBUG.

        The only log shown outside of `DEBUG` is:

        ```
        Tue Oct 18 15:14:05 UTC 2022 /tmp/shuffle-1666106044--1494069052.alert http://172.18.0.1:3001/api/v1/hooks/webhook_e19db457-e06e-4595-b5a4-f1c4d007ed70 > /dev/null 2>&
        ```

        which is quite confusing to the user and of little use. In summary, it is proposed:

        - Invalid configuration. `WARNING` log in `ossec.log`
        - Integration failure of any kind. `WARNING` level in `integrations.log`

    - ***(2)* Invalid `<level>` validation (cc @wazuh/framework)**. 
        The use of an invalid level in the integration module, like `111111111111`, should produce an error at startup time. This is the default behavior for other invalid cases like `TestingLevel` or `-11`.

    - ***(3)* Invalid `<rule_id>` validation (cc @wazuh/framework)**.
       - If string value, all alerts will be sent without applying the filter. This should be indicated to the user with a WARNING or by causing the module not to start due to a configuration error.
       - If invalid int, no alert will be sent. This should be indicated to the user with a WARNING or by causing the module not to start due to a configuration error.
       - If invalid rule in list, the valid one will be sent and the invalid one will not. Here the user should be warned that one of them is invalid.

    - ***(4)* Invalid `<group>` validation (cc @wazuh/framework)**.
       - In case of using a not defined group, no alerts passed the filter. This is the expected behaviour. However, no warning were produced.
       - In case of using a list of groups, combining defined and undefined groups, no rules passed the filter, even those from the valid groups.  Integratord should produce a warning about the invalid groups and ignoring them, considering only the valid groups (behavior different from `<rule_id>`). 

    - ***(5)* Documentation does not clearly specify that the integration optional filters are applied as an OR operation instead of an AND. (cc @wazuh/content)**

This will be discussed with the development team, and the PR will be approved or not on this basis.

**Comment by jmv74211** [comment]
## Closing conclusion 👍🏼  

|| |
|-|-|
| 🔵 | Proposed to be fixed in future versions or developments | 


The development has been approved taking into account the following considerations proposed in the [QA review](https://github.com/wazuh/wazuh-qa/issues/3398#issuecomment-1283734462):

**(1) There is not enough logging to inform the user about possible failure cases or warnings that the integration is not working properly, even if it is a configuration or suffle part. (cc @wazuh/framework)** 🔵 

This is common to all integrations, so will be fixed in future versions. A generic issue has been opened to report the need to improve logging in each of them, using this case as an example. Reported in [wazuh#15205](https://github.com/wazuh/wazuh/issues/15205)

**(2) Invalid <level> validation (cc @wazuh/framework).** 🔵 

This is common to all integrations, so will be fixed in future versions. A generic issue has been opened to report the need of improve the integrations filters validation. Reported in [wazuh#15205](https://github.com/wazuh/wazuh/issues/15205)

**(3) Invalid <rule_id> validation (cc @wazuh/framework).** 🔵 

This is common to all integrations, so will be fixed in future versions. Reported in [wazuh#15205](https://github.com/wazuh/wazuh/issues/15205)

**(4) Invalid <group> validation (cc @wazuh/framework).** 🔵 

This is common to all integrations, so will be fixed in future versions. Reported in [wazuh-documentation#15204](https://github.com/wazuh/wazuh/issues/15204)

**(5) Documentation does not clearly specify that the integration optional filters are applied as an OR operation instead of an AND. (cc @wazuh/content).** 🔵

The framework team is working on improving this documentation. It is reported at [wazuh-documentation#5677](https://github.com/wazuh/wazuh-documentation/issues/5677) and will be fixed for `4.4`, although it is not blocking this development as it is common to all integrations.
